Arrowgene.Ddon.Shared/Model/GPValidCourse.cs

```diff
@@ -0,0 +1,14 @@
+namespace Arrowgene.Ddon.Shared.Model
+{
+    public class GPValidCourse
+    {
+        public GPValidCourse()
+        {
+        }
+
+        public uint Id { get; set; }
+        public ulong StartTime { get; set; }
+        public ulong EndTime { get; set; }
```

Comment:
Should the start and end times be part of the course definition itself? Wouldn't that depend on when the character "acquires" them?

Reply:
I currently have it organized such that the courses are global to all players. The `valid_courses` array contains a list of all courses which are currently active/valid. Based on my analysis of the packets, it seems the login server always sends a list of all courses which are possible and the map of effect ids and their parameters. Later on when you select the character, it then asks for which courses are available/active for that character.

I suppose once we get the GP shop menu to work properly, we could make it such that only purchased courses will be visible for the character (I think that is how they can be obtained?). Then we can store that data in the DB.

Reply:
I think it'd be more appropriate to have the json defining the courses without start/end times, and have them hardcoded in the character select packet instead. This way the JSON contents could be the same now and when the character's courses are moved to the DB